pallets/runtime/common/src/bridge.rs

```diff
@@ -425,83 +430,115 @@ decl_module! {
         }
 
         /// Forces handling a transaction by bypassing the bridge limit and timelock.
-        #[weight = SimpleDispatchInfo::FixedOperational(20_000)]
+        #[weight = SimpleDispatchInfo::FixedOperational(250_000)]
         pub fn force_handle_bridge_tx(origin, bridge_tx: BridgeTx<T::AccountId, T::Balance>) -> DispatchResult {
             // NB: To avoid code duplication, this uses a hacky approach of temporarily whitelisting the did
             let sender = ensure_signed(origin)?;
             ensure!(sender == Self::admin(), Error::<T>::BadAdmin);
-            if let Some(did) = T::CddChecker::get_key_cdd_did(&AccountKey::try_from(bridge_tx.recipient.clone().encode())?) {
-                if !Self::bridge_whitelist(did) {
-                    // Whitelist the did temporarily
-                    <BridgeLimitWhitelist>::insert(did, true);
-                    Self::handle_bridge_tx_now(bridge_tx, false)?;
-                    <BridgeLimitWhitelist>::insert(did, false);
-                } else {
-                    // Already whitelisted
-                    return Self::handle_bridge_tx_now(bridge_tx, false);
+            Self::force_handle_signed_bridge_tx(&sender, bridge_tx)
+        }
+
+        /// Forces handling a vector of transactions by bypassing the bridge limit and timelock.
+        /// The vector is processed until the first proposal which causes an error, in which case
+        /// the error is returned and the rest of proposals are not processed.
+        ///
+        /// # Weight
+        /// `50_000 + 200_000 * bridge_txs.len()`
+        #[weight = FunctionOf(
+            |(bridge_txs,): (
+                &Vec<BridgeTx<T::AccountId, T::Balance>>,
+            )| {
+                50_000 + 200_000 * u32::try_from(bridge_txs.len()).unwrap_or_default()
+            },
+            DispatchClass::Operational,
+            true
+        )]
+        pub fn force_handle_bridge_txs(origin, bridge_txs: Vec<BridgeTx<T::AccountId, T::Balance>>) ->
+            DispatchResult
+        {
+            let sender = ensure_signed(origin)?;
+            ensure!(sender == Self::admin(), Error::<T>::BadAdmin);
+            let mut failed_txs = Vec::new();
+            for tx in bridge_txs {
+                if let Err(e) = Self::force_handle_signed_bridge_tx(&sender, tx.clone()) {
+                    failed_txs.push((tx, e.encode()));
```

Comment:
emitting just the event id should be sufficient for our use case. I don't think it's worth emitting the full tx.